packages/concerto-core/lib/decoratormanager.js

```diff
@@ -296,12 +297,57 @@ class DecoratorManager {
      */
     static executeCommand(namespace, declaration, command) {
         const { target, decorator, type } = command;
-        const { name } = ModelUtil.parseNamespace(namespace);
-        if (
-            this.falsyOrEqual(target.namespace, [namespace, name]) &&
-            this.falsyOrEqual(target.declaration, [declaration.name])
-        ) {
-            if (!target.property && !target.type) {
+        const { name } = ModelUtil.parseNamespace( namespace );
```

Comment:
This seems like a lot compared to the existing code... 50 lines to add decorators to map key/value. No way to simplify?

Reply:
I've isolated logic to `applyDecoratorForMapElement()`